entity_parser.go

```diff
@@ -40,141 +39,10 @@ const (
 )
 
 var (
-	primaryKeyPattern0 = regexp.MustCompile(`primaryKey\s*=\s*([^=]*)((\s+.*=)|$)`)
-	primaryKeyPattern1 = regexp.MustCompile(`\(\s*\((.*)\)(.*)\)`)
-	primaryKeyPattern2 = regexp.MustCompile(`\(\s*([^,\s]+),?(.*)\)`)
-	primaryKeyPattern3 = regexp.MustCompile(`^\s*([^(),\s]+)\s*$`)
-
-	indexKeyPattern0 = regexp.MustCompile(`key\s*=\s*([^=]*)((\s+.*=)|$)`)
-
-	namePattern0 = regexp.MustCompile(`name\s*=\s*(\S*)`)
-
-	columnsPattern = regexp.MustCompile(`columns\s*=\s*\(([^\(\)]+)\)`)
-
-	etlPattern0 = regexp.MustCompile(`etl\s*=\s*(\S*)`)
-
-	ttlPattern0 = regexp.MustCompile(`ttl\s*=\s*(\S*)`)
-
 	indexType = reflect.TypeOf((*Index)(nil)).Elem()
+	tagsKey   = []string{"primaryKey", "key", "name", "columns", "etl", "ttl"}
```

Comment:
I see the same strings are repeated in the below code, which can be error-prone. How about defining them as variables like:

primaryKeyTag = "primaryKey";

and use "primaryKeyTag" in the remaining places so that compiler will tell you whether or not you are spelling correctly.